### PR TITLE
reset socket on all timeouts

### PR DESCRIPTION
**Pull request by kkroo**
WIP.

If you send before getting a response from a previous send you get:

```
ZMQError: Operation cannot be accomplished in current state
```

### CODE REVIEW

**Comment by kkroo** [comment]
PTAL

**Comment by yosemitebandit** [comment]
can you add a short note in the readme describing this new version?

**Comment by kkroo** [comment]
readme added

**Comment by yosemitebandit** [comment]
will take another look once you tame the codeship :p

**Comment by kkroo** [comment]
codeship tamed!

File: openbts/core.py [hunk 1]
```diff
@@ -208,4 +207,4 @@ def __init__(self, raw_response_data):
         raise InvalidRequestError('unknown action')
     # Handle unknown response codes.
     else:
-      raise InvalidResponseError('code "%s" not known' % data['code'])
+      raise InvalidResponseError('code not known' % data['code'])
```
Comment (yosemitebandit) [inline openbts/core.py]: this statement is missing a `%s` in the string?

Comment (kkroo) [inline openbts/core.py]: thats an artifact, fixed

File: openbts/core.py [hunk 2]
```diff
@@ -151,14 +151,13 @@ def _send_and_receive(self, message):
...
       except zmq.Again:
-        # Reset the socket or it will be left in a bad state, waiting for a
-        # response.
-        self.socket.close()
-        self.setup_socket()
-        self.socket.connect(self.address)
-        raise TimeoutError('recv did not receive a response')
-    else:
-      raise TimeoutError('polling did not receive a response')
+          pass
```
Comment (yosemitebandit) [inline openbts/core.py]: nit: two space tabs here

Comment (kkroo) [inline openbts/core.py]: fixed

Comment (kkroo) [inline openbts/core.py]: looks like there is a few more. auto-indenting at 4. my bad

File: openbts/core.py [hunk 3]
```diff
@@ -151,14 +151,13 @@ def _send_and_receive(self, message):
...
-        # response.
-        self.socket.close()
-        self.setup_socket()
-        self.socket.connect(self.address)
-        raise TimeoutError('recv did not receive a response')
-    else:
-      raise TimeoutError('polling did not receive a response')
+        pass
+    # Reset the socket or it will be left in a bad state, waiting for a
+    # response.
```
Comment (yosemitebandit) [inline openbts/core.py]: uber nit: can you add to this comment that we are now closing the socket if polling fails _or_ if `recv` fails

**Comment by yosemitebandit** [comment]
LGTM

File: openbts/core.py (hunk 3 above)
Comment (kkroo) [inline openbts/core.py]: sure, done

**Pull request merged**